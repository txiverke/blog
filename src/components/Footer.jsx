// @flow

import React from 'react'

const socialMedia = [
  { name: 'Twitter', url: 'https://twitter.com/txiverke' },
  { name: 'Github', url: 'https://github.com/txiverke' },
  { name: 'Instagram', url: 'https://www.instagram.com/txiverke/' },
]

const Footer = () => (
  <footer className="app-footer">
    <ul className="app-footer-list">
    {socialMedia.map(item => 
      <li key={item.name} className="app-footer-item">
        <a href={item.url} className={`icon-${item.name.toLowerCase()}`} target="_blank">
          <span>{`Link to my ${item.name}`}</span>
        </a>
      </li>
    )}
    </ul>
  </footer>
)

export default Footer