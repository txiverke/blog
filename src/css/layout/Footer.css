.app-footer {
  position: relative;
  bottom: 0;
  z-index: 1000;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding-bottom: 0.5rem;
}

.app-footer-bg {
  position: absolute;
  background: #000000;
  width: 100%;
  height: 100vh;
  opacity: 0.4;
}

.app-footer-list{
  flex: 1;
  list-style: none;
  display: flex;
  flex-direction: row;
  justify-content: center;
  width: 100%;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.app-footer-item {
  margin: 0.5rem 1rem 0.25rem;
}

.app-footer-item a{
  font-size: var(--textSize);
  color: var(--textColor);
  text-decoration: none;
}

.app-footer span {
  display: none;
}