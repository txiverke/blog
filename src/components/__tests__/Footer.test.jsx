import React from 'react'
import Footer from '../Footer'
import { shallow } from 'enzyme'

describe('<Footer />', () => {
  xit('should render itself', () => {
    const year = new Date()
    const component = shallow(<Footer year={year} />)
    expect(component).toMatchSnapshot()
  })
})