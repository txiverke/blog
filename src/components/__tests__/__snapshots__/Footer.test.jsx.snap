// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    year={2018-03-02T18:24:38.491Z}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul
          className="app-footer-list"
>
          <li
                    className="app-footer-item"
          >
                    <a
                              className="icon-twitter"
                              href="https://twitter.com/txiverke"
                              target="_blank"
                    >
                              <span>
                                        Link to my Twitter
                              </span>
                    </a>
          </li>
          <li
                    className="app-footer-item"
          >
                    <a
                              className="icon-github"
                              href="https://github.com/txiverke"
                              target="_blank"
                    >
                              <span>
                                        Link to my Github
                              </span>
                    </a>
          </li>
          <li
                    className="app-footer-item"
          >
                    <a
                              className="icon-instagram"
                              href="https://www.instagram.com/txiverke/"
                              target="_blank"
                    >
                              <span>
                                        Link to my Instagram
                              </span>
                    </a>
          </li>
</ul>,
        <Link
          className="txt-tiny txt-transparent"
          replace={false}
          to="/admin"
>
          ©  
          <time
                    dateTime={2018-03-02T18:24:38.491Z}
          >
                    2018
          </time>
           xaviervila.tech
</Link>,
      ],
      "className": "app-footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="app-footer-item"
>
              <a
                            className="icon-twitter"
                            href="https://twitter.com/txiverke"
                            target="_blank"
              >
                            <span>
                                          Link to my Twitter
                            </span>
              </a>
</li>,
            <li
              className="app-footer-item"
>
              <a
                            className="icon-github"
                            href="https://github.com/txiverke"
                            target="_blank"
              >
                            <span>
                                          Link to my Github
                            </span>
              </a>
</li>,
            <li
              className="app-footer-item"
>
              <a
                            className="icon-instagram"
                            href="https://www.instagram.com/txiverke/"
                            target="_blank"
              >
                            <span>
                                          Link to my Instagram
                            </span>
              </a>
</li>,
          ],
          "className": "app-footer-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Twitter",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="icon-twitter"
                href="https://twitter.com/txiverke"
                target="_blank"
>
                <span>
                                Link to my Twitter
                </span>
</a>,
              "className": "app-footer-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Link to my Twitter
</span>,
                "className": "icon-twitter",
                "href": "https://twitter.com/txiverke",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Link to my Twitter",
                },
                "ref": null,
                "rendered": "Link to my Twitter",
                "type": "span",
              },
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Github",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="icon-github"
                href="https://github.com/txiverke"
                target="_blank"
>
                <span>
                                Link to my Github
                </span>
</a>,
              "className": "app-footer-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Link to my Github
</span>,
                "className": "icon-github",
                "href": "https://github.com/txiverke",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Link to my Github",
                },
                "ref": null,
                "rendered": "Link to my Github",
                "type": "span",
              },
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Instagram",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="icon-instagram"
                href="https://www.instagram.com/txiverke/"
                target="_blank"
>
                <span>
                                Link to my Instagram
                </span>
</a>,
              "className": "app-footer-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Link to my Instagram
</span>,
                "className": "icon-instagram",
                "href": "https://www.instagram.com/txiverke/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Link to my Instagram",
                },
                "ref": null,
                "rendered": "Link to my Instagram",
                "type": "span",
              },
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "©  ",
            <time
              dateTime={2018-03-02T18:24:38.491Z}
>
              2018
</time>,
            " xaviervila.tech",
          ],
          "className": "txt-tiny txt-transparent",
          "replace": false,
          "to": "/admin",
        },
        "ref": null,
        "rendered": Array [
          "©  ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2018,
              "dateTime": 2018-03-02T18:24:38.491Z,
            },
            "ref": null,
            "rendered": 2018,
            "type": "time",
          },
          " xaviervila.tech",
        ],
        "type": [Function],
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="app-footer-list"
>
            <li
                        className="app-footer-item"
            >
                        <a
                                    className="icon-twitter"
                                    href="https://twitter.com/txiverke"
                                    target="_blank"
                        >
                                    <span>
                                                Link to my Twitter
                                    </span>
                        </a>
            </li>
            <li
                        className="app-footer-item"
            >
                        <a
                                    className="icon-github"
                                    href="https://github.com/txiverke"
                                    target="_blank"
                        >
                                    <span>
                                                Link to my Github
                                    </span>
                        </a>
            </li>
            <li
                        className="app-footer-item"
            >
                        <a
                                    className="icon-instagram"
                                    href="https://www.instagram.com/txiverke/"
                                    target="_blank"
                        >
                                    <span>
                                                Link to my Instagram
                                    </span>
                        </a>
            </li>
</ul>,
          <Link
            className="txt-tiny txt-transparent"
            replace={false}
            to="/admin"
>
            ©  
            <time
                        dateTime={2018-03-02T18:24:38.491Z}
            >
                        2018
            </time>
             xaviervila.tech
</Link>,
        ],
        "className": "app-footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="app-footer-item"
>
                <a
                                className="icon-twitter"
                                href="https://twitter.com/txiverke"
                                target="_blank"
                >
                                <span>
                                                Link to my Twitter
                                </span>
                </a>
</li>,
              <li
                className="app-footer-item"
>
                <a
                                className="icon-github"
                                href="https://github.com/txiverke"
                                target="_blank"
                >
                                <span>
                                                Link to my Github
                                </span>
                </a>
</li>,
              <li
                className="app-footer-item"
>
                <a
                                className="icon-instagram"
                                href="https://www.instagram.com/txiverke/"
                                target="_blank"
                >
                                <span>
                                                Link to my Instagram
                                </span>
                </a>
</li>,
            ],
            "className": "app-footer-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Twitter",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="icon-twitter"
                  href="https://twitter.com/txiverke"
                  target="_blank"
>
                  <span>
                                    Link to my Twitter
                  </span>
</a>,
                "className": "app-footer-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Link to my Twitter
</span>,
                  "className": "icon-twitter",
                  "href": "https://twitter.com/txiverke",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Link to my Twitter",
                  },
                  "ref": null,
                  "rendered": "Link to my Twitter",
                  "type": "span",
                },
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Github",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="icon-github"
                  href="https://github.com/txiverke"
                  target="_blank"
>
                  <span>
                                    Link to my Github
                  </span>
</a>,
                "className": "app-footer-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Link to my Github
</span>,
                  "className": "icon-github",
                  "href": "https://github.com/txiverke",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Link to my Github",
                  },
                  "ref": null,
                  "rendered": "Link to my Github",
                  "type": "span",
                },
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Instagram",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="icon-instagram"
                  href="https://www.instagram.com/txiverke/"
                  target="_blank"
>
                  <span>
                                    Link to my Instagram
                  </span>
</a>,
                "className": "app-footer-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Link to my Instagram
</span>,
                  "className": "icon-instagram",
                  "href": "https://www.instagram.com/txiverke/",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Link to my Instagram",
                  },
                  "ref": null,
                  "rendered": "Link to my Instagram",
                  "type": "span",
                },
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "©  ",
              <time
                dateTime={2018-03-02T18:24:38.491Z}
>
                2018
</time>,
              " xaviervila.tech",
            ],
            "className": "txt-tiny txt-transparent",
            "replace": false,
            "to": "/admin",
          },
          "ref": null,
          "rendered": Array [
            "©  ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2018,
                "dateTime": 2018-03-02T18:24:38.491Z,
              },
              "ref": null,
              "rendered": 2018,
              "type": "time",
            },
            " xaviervila.tech",
          ],
          "type": [Function],
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
